src/controllers/profile.js

```diff
@@ -0,0 +1,140 @@
+import user from '../db/models';
+
+const { User } = user;
+
+/**
+ * @description class representing user profiles
+ *
+ * @class UserProfileHandler
+ */
+class userProfileHandler {
+  /**
+   * @description - This method is responsible for retrieving a currently logged in user
+   *
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   *
+   * @returns {object} - object representing response message
+   *
+   * @memberof UserProfileHandler
+   */
+  static async getCurrentUser(request, response) {
+    const { payload } = request.userData;
+    const userId = payload.id;
+    try {
+      const foundUser = await User.findOne({
+        where: { id: userId }
+      });
+
+      const currentUser = {
+        username: foundUser.username,
+        email: foundUser.email,
+        bio: foundUser.bio,
+        image: foundUser.image
+      };
+      if (foundUser && !foundUser.error) {
+        return response.status(200).json({
+          status: 'Success',
+          message: 'Retrieved user successfully',
+          currentUser
+        });
+      }
+    } catch (error) {
+      response.status(500).json({
+        status: 'Fail',
+        message: 'Could not retrieve user at this moment',
```

Comment:
Nice one Juwon, I would like to know why you were sending back a custom error message and not the one from the callback. You could have something like 'error: error.message'.

Reply:
Nice observation, I felt the user might not be able to understand the error message It throws, but from the small research I just did I've found it to be quite descriptive. Thanks, I'll update it now